### PR TITLE
compat API create/pull/push: fix error handling

### PR DESCRIPTION
**Pull request by vrothberg**
Please refer to the individual commits for details.  I am sure there are more such cases scattered across the compat and libpod API.  For now, I focus on reported issues to get the fixes out in time with the upcoming 4.6 release.

#### Does this PR introduce a user-facing change?

```release-note
Fix error handling in the Docker-compat API when pulling and pushing images.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by vrothberg** [comment]
@Luap99 PTAL

File: test/apiv2/10-images.at [hunk 1]
```diff
@@ -3,6 +3,7 @@
 # Tests for image-related endpoints
 #
 
+
```
Comment (vrothberg) [inline test/apiv2/10-images.at:6]: Will remove this one in case I have to repush

File: test/apiv2/70-short-names.at [hunk 2]
```diff
@@ -82,10 +82,9 @@ t DELETE "images/foo"                                200
 
 ########## PUSH
 
-t POST   "images/alpine/push?destination=localhost:9999/do:exist"                    200
-t POST   "images/quay.io/libpod/alpine/push?destination=localhost:9999/do/not:exist" 200 # Error is in the response
+t POST   "images/quay.io/libpod/alpine/push?destination=localhost:9999/do/not:exist" 500
 t POST   "images/quay.io/libpod/alpine/tag?repo=foo"                                 201
-t POST   "images/foo/push?destination=localhost:9999/do/not:exist"                   200 # Error is in the response
+t POST   "images/foo/push?destination=localhost:9999/do/not:exist"                   500
```
Comment (Luap99) [inline test/apiv2/70-short-names.at:87]: I would expect one test for the 404 case when to local image does not exits.

Comment (vrothberg) [reply test/apiv2/70-short-names.at:87]: The local images exists but the destination (registry) does not.

Comment (Luap99) [reply test/apiv2/70-short-names.at:87]: ah ok, I didn't saw the existing test.

**Review by Luap99 — COMMENTED** [review]
/lgtm
/hold

**Pull request merged**